buildmodel/testdata/.gitignore

```diff
@@ -0,0 +1,4 @@
+# This test data directory contains input/output JSON files and the wanted output. Tests also
+# generate a "*-actual.json" file when they run to make it easy for a dev to rewrite the wanted file
+# when the desired behavior changes or new cases are added. Ignore this artifact in Git.
+*-actual.json
```

Comment:
Is it common for Go testing to generate files into the `testdata` directory like this? Generally I think of using `%TEMP%` or the like for test code I/O to keep the source tree pure. In the past teams like VS would even go so far as to make the source tree directory completley read only to enforce this

Reply:
It seems pretty common. I caught a glance of this idea a while back in `gotestsum`'s readme, `-update-golden`: https://github.com/gotestyourself/gotestsum#re-running-failed-tests. I took a fresh look now, and saw a couple articles:

* https://ieftimov.com/posts/testing-in-go-golden-files/
* https://medium.com/soon-london/testing-with-golden-files-in-go-7fccc71c43d3

That tipped me off that this pattern is actually used in golang/go: https://github.com/golang/go/blob/91b9915d3f6f8cd2e9e9fda63f67772803adfa03/src/cmd/gofmt/gofmt_test.go#L18

**But** this is a little different from what I have here--going further. Passing `-update` to the test command is definitely more intentional than writing a new file upon failure. Much cleaner, and perhaps orthogonal to making the test dir readonly.

I'll change it to an `-update` flag, I like that workflow better anyway. 😄